### PR TITLE
Rewrite parse_address utils

### PR DESCRIPTION
**Pull request by hramezani**
better handling for unix socket address in `parse_address`

### CODE REVIEW

**Comment by tilgovi** [comment]
Thank you for the PR. This is definitely cleaner. It is only used by the arbiter. I had to double check that because if this were used for every connection I would worry about performance.

I think this is good, though! 👍

File: gunicorn/util.py [hunk 1]
```diff
@@ -232,11 +233,8 @@ def is_ipv6(addr):
 
 
 def parse_address(netloc, default_port=8000):
-    if netloc.startswith("unix://"):
-        return netloc.split("unix://")[1]
-
-    if netloc.startswith("unix:"):
-        return netloc.split("unix:")[1]
+    if re.match('unix:(//)?', netloc):
```
Comment (berkerpeksag) [inline gunicorn/util.py]: I'd use `r''` strings for regexes.

**Comment by hramezani** [comment]
@berkerpeksag , thanks. tests added.

**Comment by berkerpeksag** [comment]
Thanks!

**Pull request merged**